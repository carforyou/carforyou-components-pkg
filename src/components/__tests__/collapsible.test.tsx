import React from "react"
import { fireEvent, render, screen } from "@testing-library/react"

import Collapsible from "../collapsible"

const openedText = "Close me"
const closedText = "Open me"
const content = "I am content"

const renderWrapper = ({ forceCollapse = true, onChange = null } = {}) =>
  render(
    <Collapsible
      renderToggle={(isCollapsed) => (isCollapsed ? closedText : openedText)}
      onChange={onChange}
      forceCollapse={forceCollapse}
    >
      {() => content}
    </Collapsible>
  )

describe("<Collapsible>", () => {
  describe("default", () => {
    it("is closed", () => {
      renderWrapper()

      expect(screen.getByText(closedText))
      expect(screen.queryAllByText(openedText)).toEqual([])
      expect(screen.queryAllByText(content)).toEqual([])
    })

    it("can be opened", () => {
      renderWrapper()

      fireEvent.click(screen.getByText(closedText))

      expect(screen.queryAllByText(closedText)).toEqual([])
      expect(screen.getByText(openedText))
      expect(screen.getByText(content))
    })
  })

  describe("rendered open", () => {
    it("is open", () => {
      renderWrapper({
        forceCollapse: false,
      })

      expect(screen.queryAllByText(closedText)).toEqual([])
      expect(screen.getByText(openedText))
      expect(screen.getByText(content))
    })

    it("can be closed", () => {
      renderWrapper({
        forceCollapse: false,
      })

      fireEvent.click(screen.getByText(openedText))

      expect(screen.getByText(closedText))
      expect(screen.queryAllByText(openedText)).toEqual([])
      expect(screen.queryAllByText(content)).toEqual([])
    })
  })

  describe("on change function", () => {
    it("should call the onChange function if defined", () => {
      const customOnChangeFunction = jest.fn()
      renderWrapper({
        onChange: customOnChangeFunction,
      })

      fireEvent.click(screen.getByText(closedText))
      expect(customOnChangeFunction).toHaveBeenCalledWith(false)

      fireEvent.click(screen.getByText(openedText))
      expect(customOnChangeFunction).toHaveBeenCalledWith(true)
    })
  })
})
