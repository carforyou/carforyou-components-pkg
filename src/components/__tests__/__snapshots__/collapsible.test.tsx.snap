// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Collapsible> chevron variations should render mobile chevron correct 1`] = `
<div>
  <a
    class="flex items-center cursor-pointer"
    data-collapsed="true"
  >
    <div
      class="flex w-12/12 items-start lg:items-center"
    >
      Open me
      <svg
        class=""
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.6667 13.3333L16 18.6666L21.3333 13.3333"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </a>
</div>
`;

exports[`<Collapsible> chevron variations should render teal chevron correct 1`] = `
<div>
  <a
    class="flex items-center cursor-pointer"
    data-collapsed="true"
  >
    <div
      class="flex w-12/12 items-center"
    >
      Open me
      <svg
        class="text-teal"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.6667 13.3333L16 18.6666L21.3333 13.3333"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </a>
</div>
`;
